common/auto-updater/versions-db-manager.h

```diff
@@ -0,0 +1,138 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <unordered_map>
+#include <functional>
+#include <string>
+#include <vector>
+
+#include "http-downloader.h"
+
+namespace rs2
+{
+
+    class versions_db_manager
+    {
+    public:
+        struct version
+        {
+            int mjor, mnor, patch, build;
+
+            version() : mjor(0), mnor(0), patch(0), build(0) {}
+
+            version(long long ver) : version()
+            {
+                build = ver % 10000;
+                patch = (ver / 10000) % 100;
+                mnor = (ver / 1000000) % 100;
+                mjor = (ver / 100000000) % 100;
+            }
+
+            version(const std::string& str) : version()
+            {
+                std::regex rgx("^(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,5})?$");
+                std::smatch match;
+
+                if (std::regex_search(str.begin(), str.end(), match, rgx) && match.size() >= 4)
```

Comment:
Replace the hard-coded values (4,5) with constrain names for maintainability

Reply:
Agreed

Reply:
Done